### PR TITLE
Pool functional tests

### PR DESCRIPTION
**Pull request by alicelyu-px**
Testing: 
https://jenkins.pwx.dev.purestorage.com/job/Users/job/alyu/job/pxdv-dev-base/2/

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -74,3 +83,104 @@ var _ = Describe("{StoragePoolExpandSmoky}", func() {
...
+		poolToBeResized = getStoragePool(poolIDToResize)
+		storageNode, err = GetNodeWithGivenPoolID(poolIDToResize)
+		log.FailOnError(err, "Failed to get node with given pool ID")
+	})
+
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
```
Comment (lsrinivas-pure) [inline tests/basic/pool_function_test.go:104]: @alicelyu-px I have incorporated your changes to select pools for the tests in storage_pool_test.
can you reuse the tests PXRestartAddDisk and PoolAddDiskReboot

Comment (alicelyu-px) [reply tests/basic/pool_function_test.go:104]: https://jenkins.pwx.dev.purestorage.com/job/Users/job/alyu/job/pxdv-7-dev/35/console

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -74,3 +83,104 @@ var _ = Describe("{StoragePoolExpandSmoky}", func() {
...
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	It("Initiate pool expansion using add-disk and reboot node", func() {
+		StartTorpedoTest("PoolExpandDiskAddWithReboot", "Initiate pool expansion using add-disk and reboot node", nil, 51309)
```
Comment (harsh-px) [inline tests/basic/pool_function_test.go:110]: Can we rename StartTorpedoTest to something like `StartLoggingForTest` ?

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -59,12 +68,12 @@ var _ = Describe("{StoragePoolExpandSmoky}", func() {
 		verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
 	})
 
-	testName = "StoragePoolExpandDiskAuto"
-	testDescription = "Validate storage pool expansion using auto option"
-	It("select a pool that has I/O and expand it by 100 GiB with auto type. ", func() {
+	It("Select a pool that has I/O and expand it by 100 GiB with auto type. ", func() {
```
Comment (harsh-px) [inline tests/basic/pool_function_test.go:71]: Same comment as https://github.com/portworx/torpedo/pull/1673/files#r1336488163 for description

File: tests/basic/pool_function_test.go [hunk 4]
```diff
@@ -74,3 +83,104 @@ var _ = Describe("{StoragePoolExpandSmoky}", func() {
...
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	It("Initiate pool expansion using add-disk and reboot node", func() {
+		StartTorpedoTest("PoolExpandDiskAddWithReboot", "Initiate pool expansion using add-disk and reboot node", nil, 51309)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
```
Comment (harsh-px) [inline tests/basic/pool_function_test.go:111]: I see `StartTorpedoTest` already has a call to ` testrailuttils.AddRunsToMilestone`. What is the difference?

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (alicelyu-px) [reply tests/basic/pool_function_test.go:112]: Well this utility is widely used in every single torpedo test. I would make the renaming in a different ticket if we find no objection from the QA team.

File: tests/basic/pool_function_test.go (hunk 4 above)
Comment (alicelyu-px) [reply tests/basic/pool_function_test.go:113]: I would defer this to @lsrinivas-pure

File: tests/basic/pool_function_test.go [hunk 5]
```diff
@@ -1,22 +1,31 @@
...
+var err error
+
+var _ = Describe("{PoolExpandSmoky}", func() {
 	BeforeEach(func() {
-		StartTorpedoTest(testName, testDescription, nil, 0)
 		contexts = scheduleApps()
 	})
 
 	JustBeforeEach(func() {
 		poolIDToResize = pickPoolToResize()
```
[the remark this answers is not in the document]
Comment (alicelyu-px) [reply tests/basic/pool_function_test.go:29]: `poolIDToResize` is needed for every single test. I did it this way to avoid heavy code duplication.

**Review by harsh-px — APPROVED** [review]

**Pull request merged**